pyomo/contrib/mindtpy/util.py

```diff
@@ -288,6 +294,66 @@ def generate_norm_inf_objective_function(model, setpoint_model, discrete_only=Fa
     return Objective(expr=obj_blk.L_infinity_obj_var)
 
 
+def generate_lag_objective_function(model, setpoint_model, config, discrete_only=False):
+    """The function generate the objective of
+
+    Args:
+        model ([type]): [description]
+        setpoint_model ([type]): [description]
+        discrete_only (bool, optional): [description]. Defaults to False.
+    """
+    copy_var_list_values(setpoint_model.MindtPy_utils.variable_list,
+                         model.MindtPy_utils.variable_list,
+                         config)
+    temp_model = setpoint_model.clone()
+    for var in temp_model.MindtPy_utils.variable_list:
+        if var.is_integer():
+            var.unfix()
+    main_objective = next(
+        temp_model.component_data_objects(Objective, descend_into=False))
+    main_objective.activate()
+    temp_model.MindtPy_utils.deactivate()
+    TransformationFactory('core.relax_integer_vars').apply_to(temp_model)
+    # TODO: PyNumero does not support discrete variables
+    # So PyomoNLP should operate on setpoint_model
+
+    # Implementation 1
```

Comment:
This is very clean! Great job and amazing that you could use PyNumero in such a way. The issue is that we might be repeating some work here that we can save, for example when you call PyomoNLP you are generating the .nl file, which (in theory) should already exist from the solution of the NLP. Is this part "expensive" and should we try to spend some time recycling the .nl file that already exists or not?

Reply:
I find an interesting about the .nl file in Pyomo. If I call baron, the file generated is .bar. If I call gams, the file generated is .gms. If I call ipopt, the file generated is .nl.
How can we unify this with different solvers?